### PR TITLE
Ticket 8 add a comment to article comments

### PR DESCRIPTION
**Pull request by cheesenbeans**

### CODE REVIEW

File: src/components/AddAComment.jsx [hunk 1]
```diff
@@ -20,25 +20,26 @@ export default function AddAComment(article_id) {
 
   function handleSubmit(event){
     event.preventDefault();
+    console.log(commentToAdd)
```
Comment (aloush-dev) [inline src/components/AddAComment.jsx]: make sure you're removing console logs

File: src/components/AddAComment.jsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import { postComment } from "../utils/utils";
+
+export default function AddAComment(article_id) {
+  const [commentToAdd, setCommentToAdd] = useState({
+    username: "tickle122",
+    body: ""
+  });
+  const [addedComment, setAddedComment] = useState({})
+  const [errorMessage, setErrorMessage] = useState(false)
+  const [commentSuccessful, setCommentSuccessful] = useState(false)
```
Comment (aloush-dev) [inline src/components/AddAComment.jsx]: This seems like a bit of unnecessary state if you were to optimistically render your comments you wouldn't need to keep track of a successful post using state.

File: src/components/AddAComment.jsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    console.log(commentToAdd)
+    if(commentToAdd.username==="" || commentToAdd.body===""){
+        setErrorMessage(true)
+    } else {
+    postComment(article_id, commentToAdd).then((comment)=>{
+       setAddedComment(comment)
+       setCommentToAdd({username: "tickle122",
+       body: ""})
+       setCommentSuccessful(true)
+    })}
```
Comment (aloush-dev) [inline src/components/AddAComment.jsx]: Please refactor this to optimistically render the comments instead, take a look at the notes and let me know if you need pointing towards that section.

**Review by aloush-dev — CHANGES_REQUESTED** [review]
Take a look at the way your comments are ordered as newest comments are showing at the bottom of a page instead of near the top where I would expect them to be and I think your backend should order them like that.

Like where this is heading, like the idea of showing the comment that the user has posted with a green border.

Please take a look into optimisticaly rendering the comments and resubmit.

File: src/components/SingleArticle.jsx [hunk 4]
```diff
@@ -24,7 +25,7 @@ export default function SingleArticle() {
           setCommentsLoading(false);
         });
       });
-  }, []);
+  }, [commentsByArticleId]);
```
Comment (cjplatten) [inline src/components/SingleArticle.jsx]: This is causing an infinite loop in your use effect. commentsByArticleId is set within the useEffect so the useEffect triggers itself. You will see this if you open the Network tab of the inspect in your browser.

**Review by cjplatten — CHANGES_REQUESTED** [review]
This is still making an api call to render the comments and is not optimistically rendering the newly posted comment.
There also doesn't appear to be any feedback for if the comment fails to post.

**Review by HannahBurks — CHANGES_REQUESTED** [review]
Please see my requested amendment in regards to your optimistic rendering

**Review by cjplatten — CHANGES_REQUESTED** [review]
Very close! The comment is still opitimistically added to comment list if there is an error and the text stays in the comment box so it is easy to click the submit button multiple times and flood the api so i would suggest clearing the comment box after it is submitted

**Pull request merged**